test/test_other.py

```diff
@@ -14434,3 +14434,20 @@ def test_uuid(self):
   def test_wasm64_no_asan(self):
     err = self.expect_fail([EMCC, test_file('hello_world.c'), '-sMEMORY64', '-fsanitize=address'])
     self.assertContained('error: MEMORY64 does not yet work with ASAN', err)
+
+  def test_js_preprocess_pre_post(self):
+    create_file('pre.js', '''
+    'use preprocessor';
+    #if ASSERTIONS
+    console.log('assertions enabled')
+    #else
+    console.log('assertions disabled')
+    #endif
+    ''')
+    create_file('post.js', '''
+    'use preprocessor';
+    console.log({{{ POINTER_SIZE }}});
+    ''')
+    self.emcc_args += ['--pre-js', 'pre.js', '--post-js', 'post.js']
+    self.do_runf(test_file('hello_world.c'), 'assertions enabled\n4')
+    self.do_runf(test_file('hello_world.c'), 'assertions disabled\n4', emcc_args=['-sASSERTIONS=0'])
```

Comment:
Please test that the string is not in the output when optimizing.

Reply:
Good point.   Turns out tools don't strip such strings.. so its not a good fit.

I decided to go with `#preprocess` instead.  This is handy because it will immediately cause the parsing of such as JS file to file if you ever try to run it directly.

Reply:
Interesting idea. `#preprocess` is nice in that it looks like a preprocessor command anyhow.

A downside is that it doesn't parse as valid JS so editors break on it, but that's the same for the other preprocessing commands anyhow. Though not `{{{ }}}`, which makes me wonder if maybe there is a way to replace `#if` etc. with something that does parse as JS... but anyhow that's separate from this.

Reply:
Yes, @brendandahl and I have talking about this a few times.